### DISCUSSION TITLE
1.00 Clipboard Conqueror is Ready.

### DISCUSSION THREAD

**Discussion by aseichter2007** [aseichter2007/ClipboardConqueror#8] [Announcements]
I haven't heard about anything going wrong in a while. It's ready. It's doneish.   Images and text streaming need a full rebuild in C#. 

### Major Updates
- Added global keyboard listener that allows cancelling a generation in progress using the ESC key. This feature has been tested on Windows 10 systems. Please report any issues experienced on other operating systems.

### Minor Updates & Improvements
- Introduced several new prompts to enhance versatility
- Improved console verbosity for better real-time feedback
- Enhanced prompt handling eliminated a couple hidden newlines that were being added.
- Upgraded command input system to increase flexibility
- Expanded compatibility with additional inference suppliers out of the box

### Known Limitations
- Binary releases are pending due to limited development time. I'm committed to keeping the software up to date as much as possible but big changes will be very intermittent.  Send me your pull requests if you'd like to lend a hand.

If it's your first time, see the [installation guide.](https://github.com/aseichter2007/ClipboardConqueror/blob/main/Readme-Install.md)

Please reach out if you have any trouble with my software. I'd be happy to help you get it working on your system or talking to a tricky inference sever. It should connect to most anything, but setting up can be tricky.  

### Happy Holidays!
I'd really love to know how my software works for you. Thank you all for enjoying Clipboard Conqueror. 

**Full Changelog**: https://github.com/aseichter2007/ClipboardConqueror/compare/0.99999...1.00

<hr /><em>This discussion was created from the release <a href='https://github.com/aseichter2007/ClipboardConqueror/releases/tag/1.00'>1.00 Clipboard Conqueror is Ready.</a>.</em>

**Comment by rossbalmer** [comment]
I would love to report how your software works for me, only I can't get it to work! I could use a little help. I've got it working up to the point where I can get the introduction text, but I can't seem to get it talking to Ollama. I'd love to get it working.

**Comment by aseichter2007** [comment]
Hi Ross,

If the ollama server is on the default port, you should be able to quick
set the endpoint like

|||ollama|

Or

|||ollamaChat|

Try each as they access ollama differently.

If that doesn't work, make sure the url and ports look right in your
setup.js ollama endpoint configuration.

Found here, but in your local directory:
https://github.com/aseichter2007/ClipboardConqueror/blob/00253b13b055f3c2cb9e99b0c3028383209a14c5/setup.js#L119

Also, make sure you're requesting a model as expected by ollama. The
default "llama3" may be malformed or attempting to download a different
model, im not a fan of ollama and haven't tested it much.  I'm pretty sure
it supports dynamic model loading, so this could lead you to timeouts or
problems.

https://github.com/aseichter2007/ClipboardConqueror/blob/00253b13b055f3c2cb9e99b0c3028383209a14c5/setup.js#L1803

If you're calling a chat endpoint ollama sets the prompt format for you,
but the standard ollama completions endpoint will need a prompt format set
in Clipboard Conqueror to match the model.

When that is all working, change the default endpoint from kobold to ollama
https://github.com/aseichter2007/ClipboardConqueror/blob/00253b13b055f3c2cb9e99b0c3028383209a14c5/setup.js#L47

More information will help me spot the trouble if this wasn't the problem.

Thank you for trying Clipboard Conqueror.

Ps. Ollama sends the generation parameters under the "options" key.  If you
constructed or edited your own endpoint, make sure you set " paramPath:
'options' " and use the ollama compliant parameter name schedule provided
in the ollama params.

It might be different for chat and completion,  it's been a long time since
I read the ollama API specs.

**Comment by aseichter2007** [comment]
From the ollama api docs:

Model names
Model names follow a model:tag format, where model can have an optional
namespace such as example/model. Some examples are orca-mini:3b-q4_1 and
llama3:70b. The tag is optional and, if not provided, will default to
latest. The tag is used to identify a specific version.

I'm pretty sure Clipboard Conqueror is trying to load a wrong model by
default.

**Comment by aseichter2007** [comment]
From the ollama api docs:

Model names
Model names follow a model:tag format, where model can have an optional
namespace such as example/model. Some examples are orca-mini:3b-q4_1 and
llama3:70b. The tag is optional and, if not provided, will default to
latest. The tag is used to identify a specific version.

I'm pretty sure Clipboard Conqueror is trying to load a wrong model by
default.

**Comment by aseichter2007** [comment]
Koboldcpp just works, contains everything you need, and way more.  Supports
images in and out. Whisper.  It does just about all I want.  One day I'm
sure the gaps will fill.

Ollama defaults to Q4 quants and handles its downloads in a way I don't
like.  It seems like every app bundles it inside rather than admitting the
app is just an API wrapper like Clipboard Conqueror.  In the end it's less
transparent to the user.

Koboldcpp you just get a model file, get the kobold exe and you can cook.
Ollama almost forces a model selector and download process where they
control the models avaliable.  I think people can upload models at will but
it's bound to be curated eventually.

You can load an arbitrary model file with ollama but it's a pain and extra
steps you need docs to discover and some command line probably.  Kobold
simply download a model that fits in your graphics card, choose the file
and go. It's defaults will get you operating.

**Comment by aseichter2007** [comment]
For a long time ollama didn't build correctly on windows but it was very
heavily recommended as it was easy to deploy containerized to remote
servers.

I have some leftover rage from those times as I'm not interested in remote
servers or virtualization so my opinions on it are colored.

Koboldcpp just works. No list of commands to get it installed. I haven't
had that experience with other local inference servers, there is always
something extra to trip over, like Clipboard Conqueror needs node.js. I
really should bundle it properly.

Ollama has come a long way since then, they're doing good work, I don't
mean to bash them.
